src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -171,7 +171,10 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
         UCT_IB_MLX5DV_SET(rst2init_qp_in, in_2init, qpn, qp->qp_num);
         UCT_IB_MLX5DV_SET(qpc, qpc, pm_state, UCT_IB_MLX5_QPC_PM_STATE_MIGRATED);
         UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.vhca_port_num, attr->super.port);
-        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.pkey_index, iface->pkey_index);
+        if (!uct_ib_iface_is_roce(iface)) {
```

Comment:
can you check in PRM if we really need to set pkey index both in create and in modify 2init?
i would expect we need it only in 2init

Reply:
yes, we need it only in modify_qp "2init".
I didn't see if we set it during creating QP (only when creating DCT)